### ISSUE TITLE
Nothing happens with .graphql files

### ISSUE THREAD

**Issue by frederikhors** [prettier/prettier-vscode#989]
### Summary

New Visual Studio Code setup. Install this extension, open a `schema.graphql` file. Nothing happens. File is still used by VS Code as "Plain text".

### Additional information

**VS Code Version**:

```
Version: 1.38.1 (system setup)
Commit: b37e54c98e1a74ba89e03073e5a3761284e3ffb0
Date: 2019-09-11T13:35:15.005Z
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.18362
```

**Prettier & Prettier Plugin Version**:

```
Name: Prettier - Code formatter
Id: esbenp.prettier-vscode
Description: VS Code plugin for prettier/prettier
Version: 2.3.0
Publisher: Esben Petersen
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode
```

**Comment by ntotten** [comment]
You need to have an extension that registers the graphql language. I tried it with this one and prettier formatted correctly. https://marketplace.visualstudio.com/items?itemName=Prisma.vscode-graphql

**Comment by robatwilliams** [comment]
I think it's easy to wrongly think (at least, it was for me), that the languages listed would be prettified after installing the extension.

Looks like GraphQL is the only non-builtin VSCode language that Prettier supports. And the only way to add a language is via an extension.

Could formatting be enabled some other way, such as based on file extension, or would it make sense for this extension to register the language as a plain text one just so it exists?

**Comment by ntotten** [comment]
@robatwilliams I have thought about registering as plain text. The other option would be to just register a stub language for graphql that didn't do anything. 

I'd need to look at what happens if we register a language and then somebody also installs the graphql extension.

I guess the question though is why wouldn't you want to install the graphql extension if you are using graphql? It would provide syntax highlighting, etc.

**Comment by robatwilliams** [comment]
I sure could install one of the many GraphQL extensions, for my small side-project I didn't need to though until now to make this extension work. I'll try the Prisma one you linked, it says though that I need to add a `.graphqlconfig` and also install Watchman - more things I don't need (maybe it'll do enough without those though).

I'm not too concerned... my main point is that the extension says it formats GraphQL, but it doesn't do that until you install another extension. Maybe it just needs a <sup>1</sup> on GraphQL with a footnote.

**Comment by ntotten** [comment]
I registered a graphql stub language in order for formatting to work out of the box. This is fixed in version 3.0.6 (preview) though this 99ee434ba70c8acfde1d80d658431e4177c3fc23

Preview installer here: https://marketplace.visualstudio.com/items?itemName=Prettier.prettier-vscode
